### ISSUE TITLE
False Negative: ST_WRITE_TO_STATIC_FROM_INSTANCE_METHOD missed in non-static inner class with local variable shadowing

### ISSUE THREAD

**Issue by leemeii** [spotbugs/spotbugs#3893]
**Describe the bug**
The rule **ST_WRITE_TO_STATIC_FROM_INSTANCE_METHOD** fails to flag writes to static fields when they occur within a non-static inner class if a local variable shadows the static field name.

While the DLS_DEAD_LOCAL_STORE detector correctly identifies the assignment to the local variable, the ST detector seems to be "masked" or bypassed in this specific nested context.

**To Reproduce**
1. Use the following minimal Java code
 
```java
public class SpotBugsSTInnerLeakMini {
    private static int f;

    // Case 1: Top-level class (Correctly reported)
    void access(SpotBugsSTInnerLeakMini o) {
        f = o.f++; // -> ST triggered
    }

    class Inner {
        public Inner() {}

        // Case 2: Non-static inner class + Shadowing (False Negative)
        void access(SpotBugsSTInnerLeakMini o) {
            int f = o.f++;      // -> Should report ST (FN)
        }

        // Case 3: Overloaded method in inner class (False Negative)
        void access(String s) {
            int f = SpotBugsSTInnerLeakMini.f++;   // -> Should report ST (FN)
        }
    }
}
```
3. Compile the code: javac SpotBugsSTInnerLeakMini.java
4. Run SpotBugs: spotbugs -textui -low -effort:max  .
5. See error

**Expected behavior**
Both the access method in the top-level class AND the methods in the Inner class should trigger the ST_WRITE_TO_STATIC_FROM_INSTANCE_METHOD warning.

**Actual behavior**
Top-level class: Correctly reports ST .

Inner class: Completely misses ST. It only reports DLS .

**Environment**
- Java version:  JDK 17
- SpotBugs Version: 4.9.8
- How do you use SpotBugs: CLI
